watcher/watcher.go

```diff
@@ -224,8 +315,12 @@ func (p *Project) cmd(exit chan bool) {
 	}()
 }
 
+type watcher interface {
+	Add(path string) error
+}
+
 // Walks the file tree of a project
-func (p *Project) walks(watcher *fsnotify.Watcher) error {
+func (p *Project) walks(watcher watcher) error {
```

Comment:
we're only using `Add(string) error` here, so no need to have a hard dependency to fsnotify.Watcher! this allows us to prepare the file walker